src/dist.rs

```diff
@@ -0,0 +1,321 @@
+    #[clap(long)]
+    pub features: Vec<String>,
+    /// Activate all available features.
+    #[clap(long)]
+    pub all_features: bool,
+    /// Do not activate the `default` features.
+    #[clap(long)]
+    pub no_default_features: bool,
+    /// Use verbose output
+    #[clap(short, long)]
+    pub verbose: bool,
+    /// Coloring: auto, always, never.
+    #[clap(long)]
+    pub color: Option<String>,
+    /// Require Cargo.lock and cache are up to date.
+    #[clap(long)]
+    pub frozen: bool,
+    /// Require Cargo.lock is up to date.
+    #[clap(long)]
+    pub locked: bool,
+    /// Run without accessing the network.
+    #[clap(long)]
+    pub offline: bool,
+    /// Ignore `rust-version` specification in packages.
+    #[clap(long)]
+    pub ignore_rust_version: bool,
+
+    /// Command passed to the build process.
+    #[clap(skip = default_build_command())]
+    pub build_command: process::Command,
+    /// Directory of all generated artifacts.
+    #[clap(skip)]
+    pub dist_dir_path: Option<PathBuf>,
+    /// Directory of all static artifacts.
+    #[clap(skip)]
+    pub static_dir_path: Option<PathBuf>,
+    /// Set the resulting app name, default to `app`.
+    #[clap(skip)]
+    pub app_name: Option<String>,
+    /// Set the command's current directory as the workspace root.
+    #[clap(skip = true)]
+    pub run_in_workspace: bool,
+}
+
+impl Dist {
+    /// Set the command used by the build process.
+    ///
+    /// The default command is the result of the [`default_build_command`].
+    pub fn build_command(mut self, command: process::Command) -> Self {
+        self.build_command = command;
+        self
+    }
+
+    /// Set the directory for the generated artifacts.
+    ///
+    /// The default for debug build is `target/debug/dist` and
+    /// `target/release/dist` for the release build.
+    pub fn dist_dir_path(mut self, path: impl Into<PathBuf>) -> Self {
+        self.dist_dir_path = Some(path.into());
+        self
+    }
+
+    /// Set the directory for the static artifacts (like `index.html`).
+    pub fn static_dir_path(mut self, path: impl Into<PathBuf>) -> Self {
+        self.static_dir_path = Some(path.into());
+        self
+    }
+
+    /// Set the resulting package name.
+    ///
+    /// The default is `app`.
+    pub fn app_name(mut self, app_name: impl Into<String>) -> Self {
+        self.app_name = Some(app_name.into());
+        self
+    }
+
+    /// Set the dist process current directory as the workspace root.
+    pub fn run_in_workspace(mut self, res: bool) -> Self {
+        self.run_in_workspace = res;
+        self
+    }
+
+    /// Build the given package for Wasm.
+    ///
+    /// This will generate JS bindings via [`wasm-bindgen`](https://docs.rs/wasm-bindgen/latest/wasm_bindgen/)
+    /// and copy files from a given static directory if any to finally return
+    /// the paths of the generated artifacts with [`DistResult`].
```

Comment:
```suggestion
    /// Build the given package for Wasm.
    ///
    /// This will generate JS bindings via [`wasm-bindgen`](https://docs.rs/wasm-bindgen/latest/wasm_bindgen/)
    /// and copy files from a given static directory if any to finally return
    /// the paths of the generated artifacts with [`DistResult`].
    ///
    /// WASM optimizations can be achieved using [`WasmOpt`] if the feature `wasm-opt` is enabled.
```